### ISSUE TITLE
rclone mount failing at boot on new FreeBSD 12.1-RELEASE

### ISSUE THREAD

**Issue by RogerAirgood** [rclone/rclone#3697]
#### What is the problem you are having with rclone?
I just updated my system to FreeBSD 12.1-RELEASE. Along with this release comes some improvements to fusefs, outlined here: https://svnweb.freebsd.org/base?view=revision&revision=352351
Another notable change is the kernel module loaded is now fusefs.ko instead of fuse.ko

Here is the problem: 
At boot I mount a gdrive using @reboot in my crontab with the following command:
@reboot rclone mount -vvv --allow-other --allow-non-empty --log-file /var/log/rclone.log --fast-list --vfs-read-chunk-size-limit off --timeout 1h --daemon --daemon-timeout 0 cachedrive: /mnt/gcache

Something goes wrong and it doesn't mount correctly, an error is attached below.
All I have to do is rerun the command from my terminal and everything mounts correctly. No other changes were made, just a simple running of the command again. 

#### What is your rclone version (output from `rclone version`)
rclone v1.49.4

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
FreeBSD 12.1-RELEASE

####  Which cloud storage system are you using? (eg Google Drive)
Google Drive 

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
rclone mount -vvv --allow-other --allow-non-empty --log-file /var/log/rclone.log --fast-list --vfs-read-chunk-size-limit off --timeout 1h --daemon --daemon-timeout 0 cachedrive: /mnt/gcache

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

goroutine 31 [running]:
bazil.org/fuse.(*Conn).ReadRequest(0xc000098ba0, 0x162a480, 0xc00040c300, 0x1870580, 0xc0007e4580)
        /wrkdirs/usr/ports/net/rclone/work/rclone-1.49.4/vendor/bazil.org/fuse/fuse.go:1007 +0x5118
bazil.org/fuse/fs.(*Server).Serve(0xc00040c300, 0x18549e0, 0xc0000b3900, 0x0, 0x0)
        /wrkdirs/usr/ports/net/rclone/work/rclone-1.49.4/vendor/bazil.org/fuse/fs/serve.go:414 +0x372
github.com/rclone/rclone/cmd/mount.mount.func1(0xc00040c300, 0xc0000b3900, 0xc000098ba0, 0xc000099020)
        /wrkdirs/usr/ports/net/rclone/work/rclone-1.49.4/cmd/mount/mount.go:98 +0x45
created by github.com/rclone/rclone/cmd/mount.mount
        /wrkdirs/usr/ports/net/rclone/work/rclone-1.49.4/cmd/mount/mount.go:97 +0x2ee

**Comment by RogerAirgood** [comment]
Update:
After a short while, the same error is outputted and mount is destroyed.

**Comment by terorie** [comment]
Please try v1.50, error might be related to #3610

**Comment by RogerAirgood** [comment]
Same result

**Comment by ncw** [comment]
Can you post the complete log?  What you posted doesn't actually show the problem :-(

Does running rclone work OK when not started from the boot sequence? What if you try it from a terminal windo without the `--daemon` flag?

**Comment by RogerAirgood** [comment]
The result is the same run from the terminal or as a daemon. It seems to mount fine but as soon as there is activity, say an md5 of a file (just as a test) it crashes. Here is a complete log, running v1.50.1-017-g0eaf5475-beta

I ls'd the directory and then ran md5 Functions.7z (just a random file in there) 

2019/11/05 15:59:58 DEBUG : rclone: Version "v1.50.1-017-g0eaf5475-beta" starting with parameters ["./rclone" "mount" "-vvv" "--allow-other" "--allow-non-empty" "--log-file" "/var/log/rclone.log" "--fast-list" "--vfs-read-chunk-size-limit" "off" "--timeout" "1h" "--daemon-timeout" "0" "cachedrive:" "/mnt/gcache"]
2019/11/05 15:59:58 DEBUG : Using config file from "/root/.config/rclone/rclone.conf"
2019/11/05 15:59:58 DEBUG : cachedrive: wrapped gdrive: at root
2019/11/05 15:59:58 INFO  : cachedrive: Cache DB path: /root/.cache/rclone/cache-backend/cachedrive.db
2019/11/05 15:59:58 INFO  : cachedrive: Cache chunk path: /root/.cache/rclone/cache-backend/cachedrive
2019/11/05 15:59:58 INFO  : cachedrive: Chunk Memory: true
2019/11/05 15:59:58 INFO  : cachedrive: Chunk Size: 10M
2019/11/05 15:59:58 INFO  : cachedrive: Chunk Total Size: 15G
2019/11/05 15:59:58 INFO  : cachedrive: Chunk Clean Interval: 1m0s
2019/11/05 15:59:58 INFO  : cachedrive: Workers: 4
2019/11/05 15:59:58 INFO  : cachedrive: File Age: 2d
2019/11/05 15:59:58 DEBUG : Adding path "cache/expire" to remote control registry
2019/11/05 15:59:58 DEBUG : Adding path "cache/stats" to remote control registry
2019/11/05 15:59:58 DEBUG : Adding path "cache/fetch" to remote control registry
2019/11/05 15:59:58 DEBUG : Cache remote cachedrive:: Mounting on "/mnt/gcache"
2019/11/05 15:59:58 DEBUG : Cache remote cachedrive:: subscribing to ChangeNotify
2019/11/05 15:59:58 DEBUG : Adding path "vfs/forget" to remote control registry
2019/11/05 15:59:58 DEBUG : Adding path "vfs/refresh" to remote control registry
2019/11/05 15:59:58 DEBUG : Adding path "vfs/poll-interval" to remote control registry
2019/11/05 15:59:58 DEBUG : : Root:
2019/11/05 15:59:58 DEBUG : : >Root: node=/, err=<nil>
2019/11/05 15:59:58 DEBUG : gdrive: Loaded invalid token from config file - ignoring
2019/11/05 15:59:59 DEBUG : gdrive: Saved new token in config file
2019/11/05 16:00:19 DEBUG : /: Attr:
2019/11/05 16:00:19 DEBUG : /: >Attr: attr=valid=1s ino=0 size=0 mode=drwxr-xr-x, err=<nil>
2019/11/05 16:00:19 DEBUG : : Statfs:
2019/11/05 16:00:20 DEBUG : : >Statfs: stat={Blocks:274877906944 Bfree:273663663445 Bavail:274877906944 Files:1000000000 Ffree:1000000000 Bsize:4096 Namelen:255 Frsize:4096}, err=<nil>
2019/11/05 16:00:20 DEBUG : /: ReadDirAll:
2019/11/05 16:00:20 DEBUG : Cache remote cachedrive:: list ''
2019/11/05 16:00:20 DEBUG : : list: warm 99 from cache for: , expiring on: 2019-11-07 02:21:38.810138099 -0400 AST
2019/11/05 16:00:20 DEBUG : : list: cached entries: [redacted]
2019/11/05 16:00:20 DEBUG : /: >ReadDirAll: item=99, err=<nil>
2019/11/05 16:00:41 DEBUG : : Statfs:
2019/11/05 16:00:41 DEBUG : : >Statfs: stat={Blocks:274877906944 Bfree:273663663445 Bavail:274877906944 Files:1000000000 Ffree:1000000000 Bsize:4096 Namelen:255 Frsize:4096}, err=<nil>
2019/11/05 16:00:41 DEBUG : /: ReadDirAll:
2019/11/05 16:00:41 DEBUG : /: >ReadDirAll: item=99, err=<nil>
2019/11/05 16:00:41 DEBUG : /: Lookup: name="Functions.7z"
2019/11/05 16:00:41 DEBUG : /: >Lookup: node=Functions.7z, err=<nil>
2019/11/05 16:00:41 DEBUG : Functions.7z: Attr:
2019/11/05 16:00:41 DEBUG : Functions.7z: >Attr: a=valid=1s ino=0 size=2881237 mode=-rw-r--r--, err=<nil>
2019/11/05 16:00:41 DEBUG : Functions.7z: Open: flags=OpenReadOnly
2019/11/05 16:00:41 DEBUG : Functions.7z: Open: flags=O_RDONLY
2019/11/05 16:00:41 DEBUG : Functions.7z: >Open: fd=Functions.7z (r), err=<nil>
2019/11/05 16:00:41 DEBUG : Functions.7z: >Open: fh=&{Functions.7z (r)}, err=<nil>
2019/11/05 16:00:41 DEBUG : &{Functions.7z (r)}: Read: len=65536, offset=0
2019/11/05 16:00:41 DEBUG : Functions.7z: ChunkedReader.openRange at 0 length 134217728
2019/11/05 16:00:42 DEBUG : Functions.7z: moving offset set from 0 to 0
2019/11/05 16:00:42 DEBUG : Functions.7z: ChunkedReader.Read at 0 length 4096 chunkOffset 0 chunkSize 134217728
2019/11/05 16:00:42 DEBUG : Functions.7z: 0: chunk retry storage: 0
2019/11/05 16:00:43 DEBUG : Functions.7z: 0: chunk retry storage: 1
2019/11/05 16:00:43 DEBUG : Functions.7z: 0: chunk retry storage: 2
2019/11/05 16:00:44 DEBUG : Functions.7z: 0: chunk retry storage: 3
2019/11/05 16:00:44 DEBUG : Functions.7z: 0: chunk retry storage: 4
2019/11/05 16:00:45 DEBUG : Functions.7z: 0: chunk retry storage: 5
2019/11/05 16:00:45 DEBUG : Functions.7z: 0: chunk retry storage: 6
2019/11/05 16:00:45 DEBUG : worker-0 <Functions.7z>: partial downloaded chunk 0
2019/11/05 16:00:46 DEBUG : Functions.7z: ChunkedReader.Read at 4096 length 8192 chunkOffset 0 chunkSize 134217728
2019/11/05 16:00:46 DEBUG : Functions.7z: ChunkedReader.Read at 12288 length 16384 chunkOffset 0 chunkSize 134217728
2019/11/05 16:00:46 DEBUG : Functions.7z: ChunkedReader.Read at 28672 length 32768 chunkOffset 0 chunkSize 134217728
2019/11/05 16:00:46 DEBUG : Functions.7z: ChunkedReader.Read at 61440 length 65536 chunkOffset 0 chunkSize 134217728
2019/11/05 16:00:46 DEBUG : Functions.7z: ChunkedReader.Read at 126976 length 131072 chunkOffset 0 chunkSize 134217728
2019/11/05 16:00:46 DEBUG : &{Functions.7z (r)}: >Read: read=65536, err=<nil>
2019/11/05 16:00:46 DEBUG : Functions.7z: Attr:
2019/11/05 16:00:46 DEBUG : Functions.7z: >Attr: a=valid=1s ino=0 size=2881237 mode=-rw-r--r--, err=<nil>
panic: opBmap

goroutine 27 [running]:
github.com/rclone/rclone/vendor/bazil.org/fuse.(*Conn).ReadRequest(0xc00007a3c0, 0x16bac70, 0xc00030e480, 0x191b720, 0xc00011c040)
        /home/runner/work/rclone/src/github.com/rclone/rclone/vendor/bazil.org/fuse/fuse.go:1007 +0x5118
github.com/rclone/rclone/vendor/bazil.org/fuse/fs.(*Server).Serve(0xc00030e480, 0x18fdb40, 0xc000114660, 0x0, 0x0)
        /home/runner/work/rclone/src/github.com/rclone/rclone/vendor/bazil.org/fuse/fs/serve.go:414 +0x372
github.com/rclone/rclone/cmd/mount.mount.func1(0xc00030e480, 0xc000114660, 0xc00007a3c0, 0xc00007b0e0)
        /home/runner/work/rclone/src/github.com/rclone/rclone/cmd/mount/mount.go:98 +0x45
created by github.com/rclone/rclone/cmd/mount.mount
        /home/runner/work/rclone/src/github.com/rclone/rclone/cmd/mount/mount.go:97 +0x2ee

**Comment by ncw** [comment]
Ah, this looks like an unimplemented feature in the fuse library

https://github.com/bazil/fuse/blob/65cc252bf6691cb3c7014bcb2c8dc29de91e3a7e/fuse.go#L1006-L1007

I've reported an issue upstream: https://github.com/bazil/fuse/issues/221

Can you do another log with `-vv --debug-fuse` and post that please to help with the debugging please?

**Comment by RogerAirgood** [comment]
Sorry, I was quite busy today. Here is the log, as requested:

2019/11/06 18:23:17 DEBUG : rclone: Version "v1.50.1-017-g0eaf5475-beta" starting with parameters ["./rclone" "mount" "-vv" "--debug-fuse" "--allow-other" "--allow-non-empty" "--log-file" "/var/log/rclone.log" "--fast-list" "--vfs-read-chunk-size-limit" "off" "--daemon" "--timeout" "1h" "--daemon-timeout" "0" "cachedrive:" "/mnt/gcache"]
2019/11/06 18:23:17 DEBUG : Using config file from "/root/.config/rclone/rclone.conf"
2019/11/06 18:23:17 DEBUG : cachedrive: wrapped gdrive: at root
2019/11/06 18:23:17 INFO  : cachedrive: Cache DB path: /root/.cache/rclone/cache-backend/cachedrive.db
2019/11/06 18:23:17 INFO  : cachedrive: Cache chunk path: /root/.cache/rclone/cache-backend/cachedrive
2019/11/06 18:23:17 INFO  : cachedrive: Chunk Memory: true
2019/11/06 18:23:17 INFO  : cachedrive: Chunk Size: 10M
2019/11/06 18:23:17 INFO  : cachedrive: Chunk Total Size: 15G
2019/11/06 18:23:17 INFO  : cachedrive: Chunk Clean Interval: 1m0s
2019/11/06 18:23:17 INFO  : cachedrive: Workers: 4
2019/11/06 18:23:17 INFO  : cachedrive: File Age: 2d
2019/11/06 18:23:17 DEBUG : Adding path "cache/expire" to remote control registry
2019/11/06 18:23:17 DEBUG : Adding path "cache/stats" to remote control registry
2019/11/06 18:23:17 DEBUG : Adding path "cache/fetch" to remote control registry
2019/11/06 18:23:17 DEBUG : Cache remote cachedrive:: Mounting on "/mnt/gcache"
2019/11/06 18:23:17 DEBUG : gdrive: Loaded invalid token from config file - ignoring
2019/11/06 18:23:17 DEBUG : Cache remote cachedrive:: subscribing to ChangeNotify
2019/11/06 18:23:17 DEBUG : Adding path "vfs/forget" to remote control registry
2019/11/06 18:23:17 DEBUG : Adding path "vfs/refresh" to remote control registry
2019/11/06 18:23:17 DEBUG : Adding path "vfs/poll-interval" to remote control registry
2019/11/06 18:23:17 DEBUG : : Root:
2019/11/06 18:23:17 DEBUG : : >Root: node=/, err=<nil>
2019/11/06 18:23:17 DEBUG : gdrive: Saved new token in config file
2019/11/06 18:23:20 DEBUG : fuse: <- Getattr [ID=0x2 Node=0x1 Uid=0 Gid=0 Pid=4107] 0x0 fl=0
2019/11/06 18:23:20 DEBUG : /: Attr:
2019/11/06 18:23:20 DEBUG : /: >Attr: attr=valid=1s ino=0 size=0 mode=drwxr-xr-x, err=<nil>
2019/11/06 18:23:20 DEBUG : fuse: -> [ID=0x2] Getattr valid=1s ino=1 size=0 mode=drwxr-xr-x
2019/11/06 18:23:20 DEBUG : fuse: <- Access [ID=0x3 Node=0x1 Uid=0 Gid=0 Pid=4107] mask=0x4
2019/11/06 18:23:20 DEBUG : fuse: -> [ID=0x3] Access
2019/11/06 18:23:20 DEBUG : fuse: <- Open [ID=0x4 Node=0x1 Uid=0 Gid=0 Pid=4107] dir=true fl=OpenReadOnly
2019/11/06 18:23:20 DEBUG : fuse: -> [ID=0x4] Open 0x1 fl=0
2019/11/06 18:23:20 DEBUG : fuse: <- Statfs [ID=0x5 Node=0x1 Uid=0 Gid=0 Pid=4107]
2019/11/06 18:23:20 DEBUG : : Statfs:
2019/11/06 18:23:20 DEBUG : : >Statfs: stat={Blocks:274877906944 Bfree:273663663445 Bavail:274877906944 Files:1000000000 Ffree:1000000000 Bsize:4096 Namelen:255 Frsize:4096}, err=<nil>
2019/11/06 18:23:20 DEBUG : fuse: -> [ID=0x5] Statfs blocks=274877906944/273663663445/274877906944 files=1000000000/1000000000 bsize=4096 frsize=4096 namelen=255
2019/11/06 18:23:20 DEBUG : fuse: <- Read [ID=0x6 Node=0x1 Uid=0 Gid=0 Pid=4107] 0x1 4096 @0x0 dir=true fl=0 lock=0 ffl=OpenReadOnly
2019/11/06 18:23:20 DEBUG : /: ReadDirAll:
2019/11/06 18:23:20 DEBUG : Cache remote cachedrive:: list ''
2019/11/06 18:23:20 DEBUG : : list: warm 99 from cache for: , expiring on: 2019-11-07 02:21:38.810138099 -0400 AST
2019/11/06 18:23:20 DEBUG : : list: cached entries: [redacted]
2019/11/06 18:23:20 DEBUG : /: >ReadDirAll: item=99, err=<nil>
2019/11/06 18:23:20 DEBUG : fuse: -> [ID=0x6] Read 4096
2019/11/06 18:23:20 DEBUG : fuse: <- Read [ID=0x7 Node=0x1 Uid=0 Gid=0 Pid=4107] 0x1 4096 @0xfa8 dir=true fl=0 lock=0 ffl=OpenReadOnly
2019/11/06 18:23:20 DEBUG : fuse: -> [ID=0x7] Read 728
2019/11/06 18:23:20 DEBUG : fuse: <- Read [ID=0x8 Node=0x1 Uid=0 Gid=0 Pid=4107] 0x1 4096 @0x1280 dir=true fl=0 lock=0 ffl=OpenReadOnly
2019/11/06 18:23:20 DEBUG : fuse: -> [ID=0x8] Read 0
2019/11/06 18:23:31 DEBUG : fuse: <- Access [ID=0x9 Node=0x1 Uid=0 Gid=0 Pid=4085] mask=0x4
2019/11/06 18:23:31 DEBUG : fuse: -> [ID=0x9] Access
2019/11/06 18:23:31 DEBUG : fuse: <- Open [ID=0xa Node=0x1 Uid=0 Gid=0 Pid=4085] dir=true fl=OpenReadOnly
2019/11/06 18:23:31 DEBUG : fuse: -> [ID=0xa] Open 0x2 fl=0
2019/11/06 18:23:31 DEBUG : fuse: <- Statfs [ID=0xb Node=0x1 Uid=0 Gid=0 Pid=4085]
2019/11/06 18:23:31 DEBUG : : Statfs:
2019/11/06 18:23:31 DEBUG : : >Statfs: stat={Blocks:274877906944 Bfree:273663663445 Bavail:274877906944 Files:1000000000 Ffree:1000000000 Bsize:4096 Namelen:255 Frsize:4096}, err=<nil>
2019/11/06 18:23:31 DEBUG : fuse: -> [ID=0xb] Statfs blocks=274877906944/273663663445/274877906944 files=1000000000/1000000000 bsize=4096 frsize=4096 namelen=255
2019/11/06 18:23:31 DEBUG : fuse: <- Read [ID=0xc Node=0x1 Uid=0 Gid=0 Pid=4085] 0x2 4096 @0x0 dir=true fl=0 lock=0 ffl=OpenReadOnly
2019/11/06 18:23:31 DEBUG : /: ReadDirAll:
2019/11/06 18:23:31 DEBUG : /: >ReadDirAll: item=99, err=<nil>
2019/11/06 18:23:31 DEBUG : fuse: -> [ID=0xc] Read 4096
2019/11/06 18:23:31 DEBUG : fuse: <- Read [ID=0xd Node=0x1 Uid=0 Gid=0 Pid=4085] 0x2 4096 @0xfa8 dir=true fl=0 lock=0 ffl=OpenReadOnly
2019/11/06 18:23:31 DEBUG : fuse: -> [ID=0xd] Read 728
2019/11/06 18:23:31 DEBUG : fuse: <- Read [ID=0xe Node=0x1 Uid=0 Gid=0 Pid=4085] 0x2 4096 @0x1280 dir=true fl=0 lock=0 ffl=OpenReadOnly
2019/11/06 18:23:31 DEBUG : fuse: -> [ID=0xe] Read 0
2019/11/06 18:23:31 DEBUG : fuse: <- Access [ID=0xf Node=0x1 Uid=0 Gid=0 Pid=4085] mask=0x1
2019/11/06 18:23:31 DEBUG : fuse: -> [ID=0xf] Access
2019/11/06 18:23:31 DEBUG : fuse: <- Lookup [ID=0x10 Node=0x1 Uid=0 Gid=0 Pid=4085] "Functions.7z"
2019/11/06 18:23:31 DEBUG : /: Lookup: name="Functions.7z"
2019/11/06 18:23:31 DEBUG : /: >Lookup: node=Functions.7z, err=<nil>
2019/11/06 18:23:31 DEBUG : Functions.7z: Attr:
2019/11/06 18:23:31 DEBUG : Functions.7z: >Attr: a=valid=1s ino=0 size=2881237 mode=-rw-r--r--, err=<nil>
2019/11/06 18:23:31 DEBUG : fuse: -> [ID=0x10] Lookup 0x2 gen=0 valid=1s attr={valid=1s ino=8590050102649675312 size=2881237 mode=-rw-r--r--}
2019/11/06 18:23:31 DEBUG : fuse: <- Access [ID=0x11 Node=0x1 Uid=0 Gid=0 Pid=4109] mask=0x1
2019/11/06 18:23:31 DEBUG : fuse: -> [ID=0x11] Access
2019/11/06 18:23:31 DEBUG : fuse: <- Access [ID=0x12 Node=0x2 Uid=0 Gid=0 Pid=4109] mask=0x4
2019/11/06 18:23:31 DEBUG : fuse: -> [ID=0x12] Access
2019/11/06 18:23:31 DEBUG : fuse: <- Open [ID=0x13 Node=0x2 Uid=0 Gid=0 Pid=4109] dir=false fl=OpenReadOnly
2019/11/06 18:23:31 DEBUG : Functions.7z: Open: flags=OpenReadOnly
2019/11/06 18:23:31 DEBUG : Functions.7z: Open: flags=O_RDONLY
2019/11/06 18:23:31 DEBUG : Functions.7z: >Open: fd=Functions.7z (r), err=<nil>
2019/11/06 18:23:31 DEBUG : Functions.7z: >Open: fh=&{Functions.7z (r)}, err=<nil>
2019/11/06 18:23:31 DEBUG : fuse: -> [ID=0x13] Open 0x3 fl=0
2019/11/06 18:23:31 DEBUG : fuse: <- Read [ID=0x14 Node=0x2 Uid=0 Gid=0 Pid=4109] 0x3 65536 @0x0 dir=false fl=0 lock=0 ffl=OpenReadOnly
2019/11/06 18:23:31 DEBUG : &{Functions.7z (r)}: Read: len=65536, offset=0
2019/11/06 18:23:31 DEBUG : Functions.7z: ChunkedReader.openRange at 0 length 134217728
2019/11/06 18:23:32 DEBUG : Functions.7z: moving offset set from 0 to 0
2019/11/06 18:23:32 DEBUG : Functions.7z: ChunkedReader.Read at 0 length 4096 chunkOffset 0 chunkSize 134217728
2019/11/06 18:23:32 DEBUG : Functions.7z: ChunkedReader.Read at 4096 length 8192 chunkOffset 0 chunkSize 134217728
2019/11/06 18:23:32 DEBUG : Functions.7z: ChunkedReader.Read at 12288 length 16384 chunkOffset 0 chunkSize 134217728
2019/11/06 18:23:32 DEBUG : Functions.7z: ChunkedReader.Read at 28672 length 32768 chunkOffset 0 chunkSize 134217728
2019/11/06 18:23:32 DEBUG : Functions.7z: ChunkedReader.Read at 61440 length 65536 chunkOffset 0 chunkSize 134217728
2019/11/06 18:23:32 DEBUG : &{Functions.7z (r)}: >Read: read=65536, err=<nil>
2019/11/06 18:23:32 DEBUG : fuse: -> [ID=0x14] Read 65536
2019/11/06 18:23:32 DEBUG : Functions.7z: ChunkedReader.Read at 126976 length 131072 chunkOffset 0 chunkSize 134217728
2019/11/06 18:23:32 DEBUG : fuse: <- Getattr [ID=0x15 Node=0x2 Uid=0 Gid=0 Pid=4109] 0x0 fl=0
2019/11/06 18:23:32 DEBUG : Functions.7z: Attr:
2019/11/06 18:23:32 DEBUG : Functions.7z: >Attr: a=valid=1s ino=0 size=2881237 mode=-rw-r--r--, err=<nil>
2019/11/06 18:23:32 DEBUG : fuse: -> [ID=0x15] Getattr valid=1s ino=8590050102649675312 size=2881237 mode=-rw-r--r--
panic: opBmap

goroutine 14 [running]:
github.com/rclone/rclone/vendor/bazil.org/fuse.(*Conn).ReadRequest(0xc0000aed80, 0x16bac70, 0xc0001be0c0, 0x191b720, 0xc000120840)
        /home/runner/work/rclone/src/github.com/rclone/rclone/vendor/bazil.org/fuse/fuse.go:1007 +0x5118
github.com/rclone/rclone/vendor/bazil.org/fuse/fs.(*Server).Serve(0xc0001be0c0, 0x18fdb40, 0xc0002d2840, 0x0, 0x0)
        /home/runner/work/rclone/src/github.com/rclone/rclone/vendor/bazil.org/fuse/fs/serve.go:414 +0x372
github.com/rclone/rclone/cmd/mount.mount.func1(0xc0001be0c0, 0xc0002d2840, 0xc0000aed80, 0xc000079500)
        /home/runner/work/rclone/src/github.com/rclone/rclone/cmd/mount/mount.go:98 +0x45
created by github.com/rclone/rclone/cmd/mount.mount
        /home/runner/work/rclone/src/github.com/rclone/rclone/cmd/mount/mount.go:97 +0x2ee

**Comment by jkoopmann** [comment]
I seem to be running into a similar issue after upgrading to 12.1:

panic: opBmap

goroutine 81 [running]:
github.com/rclone/rclone/vendor/bazil.org/fuse.(*Conn).ReadRequest(0xc001b51b00, 0x16ba6a8, 0xc001b5e180, 0x191b3a0, 0xc000370000)
        /home/runner/work/rclone/src/github.com/rclone/rclone/vendor/bazil.org/fuse/fuse.go:1007 +0x5118
github.com/rclone/rclone/vendor/bazil.org/fuse/fs.(*Server).Serve(0xc001b5e180, 0x18fd500, 0xc001b62d20, 0x0, 0x0)
        /home/runner/work/rclone/src/github.com/rclone/rclone/vendor/bazil.org/fuse/fs/serve.go:414 +0x372
github.com/rclone/rclone/cmd/mount.mount.func1(0xc001b5e180, 0xc001b62d20, 0xc001b51b00, 0xc001b51e60)
        /home/runner/work/rclone/src/github.com/rclone/rclone/cmd/mount/mount.go:98 +0x45
created by github.com/rclone/rclone/cmd/mount.mount
        /home/runner/work/rclone/src/github.com/rclone/rclone/cmd/mount/mount.go:97 +0x2ee

Mounting is ok but after a few hours I seem to be getting the error, rclone dies and I have to reload. I can see you committed two fixes. Is there a beta for us to try?

**Comment by jkoopmann** [comment]
Just tried 

[root@jpknas /tmp]# rclone version
rclone v1.50.2-056-ge14d968f-beta
- os/arch: freebsd/amd64
- go version: go1.13.4

Same result. This unfortunately renders rclone mounting useless which in turn means that tons of my data is currently not accessible... :-( Anything we can do to support?

**Comment by jkoopmann** [comment]
Looks a bit as if with v1.50.1-036-g6f98e6c9-fix-freebsd-mount-beta things are stable. I probably wrongly assumed that v1.50.2 contained the patches as well. Thanks for your great work!

**Comment by ncw** [comment]
@jkoopmann I'm still waiting for upstream to merge my fix - I'll then pull into rclone and issue a beta.

**Comment by RogerAirgood** [comment]
@ncw sorry for ping. Were able to get upstream to merge your fix? I'd love to be able to update my rclone but can't really do much past the special version you were kind enough to prepare.

**Comment by ncw** [comment]
I pinged the upstream on the pull request https://github.com/bazil/fuse/pull/222

I've considering switching to an alternate library [hanwen/go-fuse](https://github.com/hanwen/go-fuse) but I just discovered that doesn't support FreeBSD.

I guess the next step would be for me to fork the project so I can add that patch.

In the mean time I rebased the patch on master so you can have a latest beta with the patch

https://beta.rclone.org/branch/v1.50.2-085-gcebc3d7c-fix-freebsd-mount-beta/ (uploaded in 15-30 mins)

**Comment by RogerAirgood** [comment]
I appreciate the effort. That repo looks like it has hasn't had any commits for years. It would be nice to have it fixed upstream in some way but I understand that it is not ideal for you. Thanks for the latest beta!

**Comment by GremL1N** [comment]
Hi. You also can fix it by port patch.
Just add to /usr/ports/net/rclone/Makefile:
```
post-patch:
        @${REINPLACE_CMD} -e 's|panic("opBmap")|goto unrecognized|' ${WRKSRC}/vendor/bazil.org/fuse/fuse.go
```
before line  `.include <bsd.port.mk>`

Complete Makefile looks like this:
```
# $FreeBSD: head/net/rclone/Makefile 518968 2019-12-03 21:50:58Z woodsb02 $

PORTNAME=       rclone
DISTVERSIONPREFIX=      v
DISTVERSION=    1.50.2
CATEGORIES=     net

MAINTAINER=     wg@FreeBSD.org
COMMENT=        Sync files to and from various cloud services

LICENSE=        MIT
LICENSE_FILE=   ${WRKSRC}/COPYING

USES=           go:modules
USE_GITHUB=     yes
GH_ACCOUNT=     ncw

PLIST_FILES=    bin/rclone

post-patch:
        @${REINPLACE_CMD} -e 's|panic("opBmap")|goto unrecognized|' ${WRKSRC}/vendor/bazil.org/fuse/fuse.go

.include <bsd.port.mk>
```
I've tested mount with two remotes (yandex disk and s3 like), woks fine last few hours.

**Comment by ncw** [comment]
It looks like a fix for this has been merged upstream :-)

Can you give this a go please? If it works I'll merge it into the beta

https://beta.rclone.org/branch/v1.50.2-094-g02c578f1-fix-3697-freebsd-mount-beta/ (uploaded in 15-30 mins)

**Comment by RogerAirgood** [comment]
Looks good here, tested on FreeBSD 12.1-RELEASE-p1. My testing methodology consisted of writing a 500M file, ls'ing some very populated directories, and md5'ing some random files. No errors in log and it went fine.

**Comment by ncw** [comment]
Thanks for testing.

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.51
